src/chainstate/stacks/db/blocks.rs

```diff
@@ -3365,14 +3367,16 @@ impl StacksChainState {
                 return Err(MemPoolRejection::BadAddressVersionByte)
         }
 
+        // todo(ludo)
+
         // 5: the paying account must have enough funds
-        if fee as u128 > payer.stx_balance {
+        if  !payer.stx_balance.can_transfer(fee as u128, 0) {
```

Comment:
This shouldn't be 0.  It should be the current block height, since the payer may have tokens that are locked that would be unlocked in the act of spending a transaction fee.

Reply:
Good catch, thanks! I had a unresolved todo a few lines above for this issues.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1850/commits/0c6c1d9dd8e47e79643c1edea6dc95de404ed262